src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -155,6 +186,7 @@ create_reference_table(PG_FUNCTION_ARGS)
 	Oid relationId = PG_GETARG_OID(0);
 	int shardCount = 1;
 	AttrNumber firstColumnAttrNumber = 1;
+	char *colocateWithTableName = "default";
```

Comment:
Even with new implementations, without reserved colocation id for reference tables, this problem may exists.

Problem is more severe in second case. Think about like this;

- I have n nodes and no tables yet. I created a reference table. with this API
- I set replication factor to n and create a distributed table with shard_count = 1, colocate_with = none. This created new colocation group. Let's say its id is 2.
- I use mark_colocation_group_default(2);
- I create new reference table. New table will be created in colocation group 2.